Source/pack.cpp

```diff
@@ -87,41 +87,54 @@ bool IsCreationFlagComboValid(uint16_t iCreateInfo)
 	const bool isPregenItem = (iCreateInfo & CF_PREGEN) != 0;
 	const bool isUsefulItem = (iCreateInfo & CF_USEFUL) == CF_USEFUL;
 
-	if (isPregenItem)
+	if (isPregenItem) {
+		// Pregen flags are discarded when an item is picked up, therefore impossible to have in the inventory
 		return false;
+	}
 	if (isUsefulItem && (iCreateInfo & ~CF_USEFUL) != 0)
 		return false;
-	if (isTownItem && hasMultipleFlags(iCreateInfo))
+	if (isTownItem && hasMultipleFlags(iCreateInfo)) {
+		// Items from town can only have 1 towner flag
 		return false;
+	}
 	return true;
 }
 
 bool IsTownItemValid(uint16_t iCreateInfo, const Player &player)
 {
 	const uint8_t level = iCreateInfo & CF_LEVEL;
 	const bool isBoyItem = (iCreateInfo & CF_BOY) != 0;
+	const uint8_t maxTownItemLevel = 30;
 
+	// Wirt items in multiplayer are equal to the level of the player, therefore they cannot exceed the max character level
 	if (isBoyItem && level <= player.getMaxCharacterLevel())
 		return true;
 
-	return level <= 30;
+	return level <= maxTownItemLevel;
 }
 
 bool IsUniqueMonsterItemValid(uint16_t iCreateInfo, uint32_t dwBuff)
 {
 	const uint8_t level = iCreateInfo & CF_LEVEL;
 	const bool isHellfireItem = (dwBuff & CF_HELLFIRE) != 0;
 
+	// Check all unique monster levels to see if they match the item level
 	for (int i = 0; UniqueMonstersData[i].mName != nullptr; i++) {
 		const auto &uniqueMonsterData = UniqueMonstersData[i];
 		const auto &uniqueMonsterLevel = static_cast<uint8_t>(MonstersData[uniqueMonsterData.mtype].level);
 
-		if (!isHellfireItem && IsAnyOf(uniqueMonsterData.mtype, MT_HORKDMN, MT_DEFILER, MT_NAKRUL)) {
-			// These monsters don't appear in Diablo
+		if (IsAnyOf(uniqueMonsterData.mtype, MT_DEFILER, MT_NAKRUL)) {
+			// These monsters don't use their mlvl for item generation
+			continue;
+		}
+
+		if (!isHellfireItem && IsAnyOf(uniqueMonsterData.mtype, MT_HORKDMN)) {
+			// This monster doesn't appear in Diablo
 			continue;
 		}

```

Comment:
Looks like Hork Demon is in the same boat as Na-Krul and Defiler. He only drops Theodore or an amulet with ilvl 26.

https://github.com/diasurgical/devilutionX/blob/4467b8424a3d0ba7cc387b487a89de2453b20db9/Source/monster.cpp#L877-L882